Barkane/Assets/Scripts/Foldable Paper/PaperJoint.cs

```diff
@@ -10,13 +10,35 @@ public class PaperJoint : MonoBehaviour
     private bool isSelected = false; //true when this is the current selected fold
     public bool showLine = false; //true when this joint or any adjacent joins are selected. Used for showing visuals and partitioning graph
     public LineRenderer lineRenderer;
+    private PaperJoint currentJoint;
+    FoldablePaper foldablePaper;
+    List<GameObject> willBeFoldedAll;
+    private bool isFirstCall = true;
+    List<PaperSqaure> willBeFoldedPaperSquares = new List<PaperSqaure>();
+
     [SerializeField] private List<PaperJoint> adjList = new List<PaperJoint>();
 
     [SerializeField] private CapsuleCollider capsuleCollider;
    // private bool JointEnabled = true; //CO: Set to false to "cut" the paper along the given joint
     public bool canFold = true; //CO: Set to false to lock the current joint in position, as if the squares were glued together
 
-    private void Start() {
+    void Update(){
```

Comment:
Unity lifecycle functions should appear in the order that they are called, so start should be before update